RxCocoa/Common/DelegateProxy.swift

```diff
@@ -201,18 +203,45 @@ open class DelegateProxy : _RXDelegateProxy {
     ///
     /// - parameter object: Object that can have assigned delegate proxy.
     /// - parameter proxy: Delegate proxy object to assign to `object`.
-    open class func assignProxy(_ proxy: AnyObject, toObject object: AnyObject) {
-        precondition(proxy.isKind(of: self.classForCoder()))
-       
+    open class func assignProxy(_ proxy: Delegate, toObject object: ParentObject) {
         objc_setAssociatedObject(object, self.delegateAssociatedObjectTag(), proxy, .OBJC_ASSOCIATION_RETAIN)
     }
     
+    
+    /// Returns designated delegate property for object.
+    ///
+    /// Objects can have multiple delegate properties.
+    ///
+    /// Each delegate property needs to have it's own type implementing `DelegateProxyType`.
+    ///
+    /// It's abstract method.
+    ///
+    /// - parameter object: Object that has delegate property.
+    /// - returns: Value of delegate property.
+    open class func currentDelegateFor(_ object: ParentObject) -> Delegate? {
```

Comment:
I'm actually not sure we need both `DelegateProxyBase` and `DelegateProxyType`. Is there some reason we have two?